Feature/issue24

julianspaeth opened the pull request:
Query the number of following projects and which of the are overdue.
Show them in the statistics panel. Overdue projects are all projects which raw data is registered since more than 12 weeks.

julianspaeth committed:
Remove static project stats

Remove static project overview stats. They should be dynamic.

See: #24

julianspaeth committed:
Implement empty project stats module

Implement empty project stats module with MVP.

See: #24

julianspaeth committed:
Query number of following projects

Get the number and show it in the statistics panel

See: #24

julianspaeth committed:
Query number of overdue projects

Get the number and show it in the statistics panel.

overdue = > 12 weeks.

See: #24

sven1103 commented on src/main/java/life/qbic/portal/ManagerUI.java:
```diff
@@ -161,22 +165,20 @@ protected void init(VaadinRequest vaadinRequest) {
+        //Init project stats
+        final ProjectsStatsModel projectsStatsModel = new ProjectsStatsModel(projectDatabase);
+        final ProjectsStatsPresenter projectsStatsPresenter = new ProjectsStatsPresenter(projectsStatsModel, projectsStatsView, openBisConnection);
+        projectsStatsPresenter.setUserID("zxmqw74");
```
Vielleicht können wir zu Beginn eine globale Variable userId definieren, damit die Zugangsdaten nicht an mehreren Stellen hardgecodet stehen.

`final String userID = "zxmqw74";`

Nachher soll ja dann über die LiferayVaadinUtils library auf den Screenname im Portal zugegriffen werden, und die ID von dort übernommen werden.

sven1103 replied:
Einfach in der pom.xml folgende Abhängigkeit eintragen: life.qbic.openbis liferayandvaadinhelpers/1.1.2/

    <dependency>
	    <groupId>life.qbic.openbis</groupId>
	    <artifactId>liferayandvaadinhelpers</artifactId>
	    <version>1.1.2</version>
    </dependency>

Da gibts ne Klasse, die dann in Liferay auf den Screenname zugreifen kann. Andreas und Chris können dir da sicher sagen wie, habe ich selber noch nicht gemacht, sollte aber n einzeiler sein.

sven1103 replied:
kannst ja ne if-else Weiche machen, und wenn kein Screenname gefunden werden kann, dann nimm deinen hart gecodeden (zum testen). In Production solls dann natürlich n Fehler werden.

sven1103 commented on src/main/java/life/qbic/portal/projectsStatsModule/ProjectsStatsModel.java:
```diff
@@ -0,0 +1,70 @@
+        followingProjects.clear();
+
+        HashMap<String, String> querySettings = new HashMap<>();
+        querySettings.put("table", "followingprojects");
```
Vielleicht kann man das noch a weng abstrahieren. bzw veränder die Methodensignatur, sodass die methode noch nen Tablename und ne userid erwartet.

sven1103 commented on src/main/java/life/qbic/portal/projectsStatsModule/ProjectsStatsModel.java:
```diff
@@ -0,0 +1,70 @@
+        projectDatabase.connectToDatabase();
+
+        HashMap<String, String> querySettings = new HashMap<>();
+        querySettings.put("table", "projectsoverview");
```
Gleiches hier wie oben. Tabellennahme in Methodensignatur.

sven1103 commented on src/main/java/life/qbic/portal/projectsStatsModule/ProjectsStatsPresenter.java:
```diff
@@ -0,0 +1,51 @@
+
+    }
+
+    private void refreshProjects() throws SQLException, WrongArgumentSettingsException {
```
welche Klasse nutzt die Methode

sven1103 reviewed (changes requested):
Nur ein paar kleine Anregungen

julianspaeth committed:
Adjust userID and improve the code

If running as a liferay portlet get the screenname, otherwise use global variable.

Get different table names from methodsignature of projectStatsModel.

Use the update function in the MasterPresenter to refresh the projectStatsView.

See: #24, #28

sven1103 reviewed (approved):
top

sven1103 commented:
Fix #24
